### ISSUE TITLE
function body indentation false positive ("indent" rule)

### ISSUE THREAD

**Issue by feross** [eslint/eslint#3173]
With this code:

```js
export function create (id, xfilter, rawType,
                        width=defaultWidth, height=defaultHeight,
                        footerHeight=defaultFooterHeight,
                        padding=defaultPadding) {
  // ... function body, indented two spaces
}
```

And this config:

```json
"indent": [2, 2, { "indentSwitchCase": true }]
```

There is an unexpected error:

```
/t.js:6:2: Expected indentation of 24 characters.
```

I expect to be able to use 2 spaces indentation for the function body. The indentation of the function parameters should not effect the body.

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by gyandeeps** [comment]
Multi-line is still a problem with `indent` rule.
Kinda related https://github.com/eslint/eslint/issues/1801

**Comment by targos** [comment]
Cross-referencing: https://github.com/nodejs/io.js/pull/2286

**Comment by thealjey** [comment]
@feross you can reformat your code slightly to make it perfectly valid with the current version of ESLint (1.1.0):

```JavaScript
export function create (id, xfilter, rawType,
                        width=defaultWidth, height=defaultHeight,
                        footerHeight=defaultFooterHeight,
                        padding=defaultPadding
) {
  // ... function body, indented two spaces
}
```

**Comment by silverwind** [comment]
@thealjey that looks too confusing imho, and I think it wouldn't comply with the `1tbs` option of the `brace-style` rule :)

@gyandeeps I hope this issue is not forgotten. If I recall correctly, it's the last issue blocking us from upgrading 0.24 to 1.x in node.

**Comment by gyandeeps** [comment]
@silverwind Whenever we put the `accepted` tag to an issue that means it is open for people to contribute. So its never forgotten, it just that sometimes it takes time and sometimes its quick.

**Comment by thealjey** [comment]
@silverwind yes, in fact, it does comply with `1tbs`
